src/backend/distributed/planner/multi_logical_optimizer.c

```diff
@@ -1271,8 +1272,19 @@ MasterExtendedOpNode(MultiExtendedOp *originalOpNode)
 		bool hasAggregates = contain_agg_clause((Node *) originalExpression);
 		if (hasAggregates)
 		{
+			AggClauseCosts aggregateCosts;
 			Node *newNode = MasterAggregateMutator((Node *) originalExpression,
 												   walkerContext);
+
+			memset(&aggregateCosts, 0, sizeof(aggregateCosts));
+
+#if PG_VERSION_NUM >= 90600
+			get_agg_clause_costs(NULL, (Node *) newNode, AGGSPLIT_SIMPLE,
+								 &aggregateCosts);
+#else
+			count_agg_clauses(NULL, (Node *) newNode, &aggregateCosts);
+#endif
+
```

Comment:
Hm. Wouldn't it be better to do this in MasterAggregateMutator() itself? Seems quite possible to move this to the tail end of MasterAggregateMutator().